sonarts-core/src/rules/preferBaseTypeRule.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { TypedSonarRuleVisitor } from "../utils/sonarAnalysis";
+import { SymbolTableBuilder } from "../symbols/builder";
+import { is } from "../utils/nodes";
+import { SymbolTable, UsageFlag } from "../symbols/table";
+
+export class Rule extends tslint.Rules.TypedRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "prefer-base-type",
+    description: "Function parameters should be declared with base types",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-3242",
+    type: "maintainability",
+    typescriptOnly: true,
+  };
+
+  public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName, program, SymbolTableBuilder.build(sourceFile, program))
+      .visit(sourceFile)
+      .getIssues();
+  }
+
+  static message(type: ts.Type, baseType: ts.Type) {
+    const baseTypeName = baseType.symbol.name;
+    if (baseTypeName === "default") {
+      return `Use the parent type here; it is a more general type than '${type.symbol.name}'.`;
```

Comment:
Since we are not sure about the relevance of this issue, IMO we should change a bit the message in order to convey a more *informative* tone, like "It might be better to use the parent type here..."